numba/np/arraymath.py

```diff
@@ -4116,11 +4116,12 @@ def impl(ap1, ap2, mode, direction):
         # For "direction", +1 to write the return values out in order 0->N
         # -1 to write them out N->0.
 
-        if not (mode == Mode.VALID or mode == Mode.FULL):
-            raise ValueError("Invalid mode")
-
         n1 = len(ap1)
         n2 = len(ap2)
+
+        if n1 < n2:
+            raise ValueError("'len(ap1)' must greate than 'len(ap2)'")
```

Comment:
```suggestion
            raise ValueError("'len(ap1)' must greater than 'len(ap2)'")
```

Also a good idea is to add a test for this error, so that we have coverage.

Reply:
As now documented by the comments, I don't see how to cover this in tests, because this should never fail when called with `np.correlate` or `np.convolve`. This check is only here to avoid future errors if one change the upstream code of `_np_correlate.imp` or else.

Reply:
See, e.g.:

https://github.com/numba/numba/blob/0c5e47ee43053d52c2e526ecd7cfe702e9c9be79/numba/np/arraymath.py#L4202-L4205

Reply:
Got it, thanks for the explanation.